src/net.cpp

```diff
@@ -555,7 +560,8 @@ CNode* CConnman::ConnectNode(CAddress addrConnect, const char *pszDest, bool fCo
                              pszDest ? pszDest : "",
                              conn_type,
                              /*inbound_onion=*/false,
-                             CNodeOptions{ .i2p_sam_session = std::move(i2p_transient_session) });
+                             CNodeOptions{ .i2p_sam_session = std::move(i2p_transient_session),
+                                           .permission_flags = permission_flags });
```

Comment:
I got a compile error with the third commit, "Accept "in" and "out" flags to -whitelist..." because these are out of order, which you fixed later, but each commit should compile.

Reply:
Done in #27114